splicemachine/spliceengine
# DB-12314 Sqlshell init .rc, file for pre-setting sqlshell settings

## Short Description
adding a .sqlshellrc that is executed first when running sqlshell

## Long Description
We search for a `.sqlshellrc` in `(CWD)/.sqlshellrc`, then `$HOME/.sqlshellrc`.
If one of those is found, we execute the commands in it. We don't run both files, CWD has precedence.

If sqlshell.sh is used with `-f script.sql` option, we first execute `.sqlshellrc`, then `script.sql`.

Note that when using `spliceengine/sqlshell.sh`, the CWD is `spliceengine/splice_machine`.
When using sqlshell standalone artifact, use `sqlshell/.sqlshellrc`.

## New i18n options
Some new options for i18n for configuring *client-side* conversion of date/time/timestamp to strings (note that this does not change SERVER-SIDE conversions):
```
set dateformat 'YYYY-MM-dd';
set timeformat 'HH:mm:ss';
set timestampformat 'yyyy-MM-dd T HH:mm:ss.SSS';
```


## How to test
When testing in spliceengine, create a `.sqlshellrc`. Otherwise, if created a sqlshell bundle, create a .sqlshellrc in same directory. To test, save this into the file
```
set dateformat dd.MM.YYYY
```

Then execute sqlshell and run `values current_date`, it should return date in dd.MM.YYYY format.
